### PR TITLE
[ Ent-Corda ] Add Firewall Float component

### PR DESCRIPTION
**Pull request by abevers**
**Changelog**
- Add Helm Charts for Firewall Float component according to [this link](https://docs.corda.net/docs/corda-enterprise/4.4/node/corda-firewall-component.html#dmz-ready-node--bridge--float)

**To be completed**
Corda Enterprise Node integration

**To be reviewed by**
@suvajit-sarkar @jagpreetsinghsasan 

**Linked issue**
#795

### CODE REVIEW

File: platforms/corda-ent/charts/float/files/firewall.conf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+silencedIPs: []
+
+enableAMQPPacketTrace: false
+artemisReconnectionIntervalMin: 5000
+artemisReconnectionIntervalMax: 60000
+politeShutdownPeriod: 1000
+p2pConfirmationWindowSize: 1048576
+auditServiceConfiguration: {
+  loggingIntervalSec: 60
+}
```
Comment (jagpreetsinghsasan) [inline platforms/corda-ent/charts/float/files/firewall.conf:37]: Add newline

File: platforms/corda-ent/charts/float/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{{- define "labels.custom" }}
+  {{ range $key, $val := $.Values.metadata.labels }}
+  {{ $key }}: {{ $val }}
+  {{ end }}
+{{- end }}
```
Comment (jagpreetsinghsasan) [inline platforms/corda-ent/charts/float/templates/_helpers.tpl:5]: Add newline

File: platforms/corda-ent/charts/float/templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  name: {{ .Values.nodeName }}-conf
+  namespace: {{ .Values.metadata.namespace }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.nodeName }}-conf
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+data:
+  firewall.conf: |+
+{{ tpl (.Files.Get "files/firewall.conf") . | indent 4 }}
```
Comment (jagpreetsinghsasan) [inline platforms/corda-ent/charts/float/templates/configmap.yaml:14]: Add newline

File: platforms/corda-ent/charts/float/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    app: {{ .Values.nodeName }}
+# we need healthCheckNodePort set to get rid of logs pollution
+{{- if (.Values.healthCheckNodePort) }}
+  healthCheckNodePort: {{ .Values.healthCheckNodePort }}
+{{- end }}  
+  ports:
+  - name: bridge-tunnel
+    protocol: TCP
+    port: {{ .Values.bridge.tunnelPort }}
+    targetPort: {{ .Values.bridge.tunnelPort }}
```
Comment (jagpreetsinghsasan) [inline platforms/corda-ent/charts/float/templates/service.yaml:24]: Add newline

File: platforms/corda-ent/charts/float/values.yaml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  readinessthreshold:
+
+# DMZ related information to configure the Float within the network
+dmz:
+  # Provide the internal IP address (DNS name preferred) for the DMZ. This is used for the Float address.
+  # e.g. internal: "dmzinternal"
+  internal:
+  # Provide the external IP address (DNS name preferred) for the DMZ. This is used to reference the Node p2p address in the Float.
+  # e.g. internal: "dmzexternal"
+  external:
```
Comment (jagpreetsinghsasan) [inline platforms/corda-ent/charts/float/values.yaml:129]: Add newline

File: platforms/corda-ent/charts/float/files/firewall.conf [hunk 6]
```diff
@@ -0,0 +1,37 @@
+firewallMode: FloatOuter
+
+# Uses /opt/corda/workspace as base directory
+networkParametersPath: "network-parameters"
+
+inboundConfig: {
+  listeningAddress: "{{ .Values.dmz.external}}:{{ .Values.node.p2pPort}}"
```
Comment (sownak) [inline platforms/corda-ent/charts/float/files/firewall.conf:7]: spaces needed "  }}"

File: platforms/corda-ent/charts/float/files/firewall.conf [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    expectedCertificateSubject: "{{ .Values.bridge.legalName }}"
+    tunnelSSLConfiguration: {
+           keyStorePassword: "password"
+           trustStorePassword: "password"
+           sslKeystore: "certificates/float.jks"
+           trustStoreFile: "certificates/trust.jks"
+    }
+}
+
+revocationConfig: { mode: "EXTERNAL_SOURCE"})
```
Comment (sownak) [inline platforms/corda-ent/charts/float/files/firewall.conf:21]: check if ending ) is needed

File: platforms/corda-ent/charts/float/values.yaml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  # E.g. readinessthreshold: 15
+  readinessthreshold:
+
+# DMZ related information to configure the Float within the network
+dmz:
+  # Provide the internal IP address (DNS name preferred) for the DMZ. This is used for the Float address.
+  # e.g. internal: "dmzinternal"
+  internal:
+  # Provide the external IP address (DNS name preferred) for the DMZ. This is used to reference the Node p2p address in the Float.
+  # e.g. internal: "dmzexternal"
```
Comment (sownak) [inline platforms/corda-ent/charts/float/values.yaml:128]: correct the example

File: platforms/corda-ent/charts/float/values.yaml (hunk 5 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/values.yaml]: Fixed in [9862bdb ](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/9862bdbebd85ac9508aaec96a1807981550cf571)

File: platforms/corda-ent/charts/float/templates/service.yaml (hunk 4 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/templates/service.yaml]: Fixed in [9862bdb ](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/9862bdbebd85ac9508aaec96a1807981550cf571)

File: platforms/corda-ent/charts/float/templates/configmap.yaml (hunk 3 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/templates/configmap.yaml]: Fixed in [9862bdb](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/9862bdbebd85ac9508aaec96a1807981550cf571)

File: platforms/corda-ent/charts/float/templates/_helpers.tpl (hunk 2 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/templates/_helpers.tpl]: Fixed in [9862bdb](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/9862bdbebd85ac9508aaec96a1807981550cf571)

File: platforms/corda-ent/charts/float/files/firewall.conf (hunk 1 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/files/firewall.conf]: Fixed in [9862bdb](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/9862bdbebd85ac9508aaec96a1807981550cf571)

File: platforms/corda-ent/charts/float/values.yaml (hunk 8 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/values.yaml]: Fixed in [356b94a](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/356b94a6b9bda311911f1dc595f0acb84792baa8)

File: platforms/corda-ent/charts/float/files/firewall.conf (hunk 7 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/files/firewall.conf]: Can be removed. Fixed in [356b94a](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/356b94a6b9bda311911f1dc595f0acb84792baa8)

File: platforms/corda-ent/charts/float/files/firewall.conf (hunk 6 above)
Comment (abevers) [reply platforms/corda-ent/charts/float/files/firewall.conf:7]: Fixed in [356b94a](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/893/commits/356b94a6b9bda311911f1dc595f0acb84792baa8)

**Pull request merged**